climada_petals/hazard/tc_tracks_forecast.py

```diff
@@ -188,80 +198,154 @@ def fetch_bufr_ftp(target_dir=None, remote_dir=None):
 
         return localfiles
 
-    def read_one_bufr_tc(self, file, id_no=None, fcast_rep=None):
-        """ Read a single BUFR TC track file.
+    def read_one_bufr_tc(self, file, id_no=None):
+        """ Read a single BUFR TC track file tailored to the ECMWF TC track
+        predictions format.
 
-        Parameters
-        ----------
-        file : str, filelike
-            Path object, string, or file-like object
-        id_no : int
-            Numerical ID; optional. Else use date + random int.
-        fcast_rep : int
-            Of the form 1xx000, indicating the delayed
-            replicator containing the forecast values; optional.
+        Parameters:
+            file (str, filelike): Path object, string, or file-like object
+            id_no (int): Numerical ID; optional. Else use date + random int.
         """
+        # Open the bufr file
+        try:
+            # for the case that file is str, try open it
+            file = open(file, 'rb')
+        except TypeError:
+            pass
+        bufr = ec.codes_bufr_new_from_file(file)
+        # we need to instruct ecCodes to expand all the descriptors
+        # i.e. unpack the data values
+        ec.codes_set(bufr, 'unpack', 1)
+
+        # get the forcast time
+        timestamp_origin = dt.datetime(
+            ec.codes_get(bufr, 'year'), ec.codes_get(bufr, 'month'),
+            ec.codes_get(bufr, 'day'), ec.codes_get(bufr, 'hour'),
+            ec.codes_get(bufr, 'minute'),
+        )
+        timestamp_origin = np.datetime64(timestamp_origin)
 
-        decoder = pybufrkit.decoder.Decoder()
+        # get storm identifier
+        sid = ec.codes_get(bufr, 'stormIdentifier').strip()
 
-        if hasattr(file, 'read'):
-            bufr = decoder.process(file.read())
-        elif hasattr(file, 'read_bytes'):
-            bufr = decoder.process(file.read_bytes())
-        elif Path(file).is_file():
-            with Path(file).open('rb') as i:
-                bufr = decoder.process(i.read())
-        else:
-            raise FileNotFoundError('Check file argument')
+        # number of timestep (size of the forecast time + initial timestep)
+        try:
+            n_timestep = ec.codes_get_size(bufr, 'timePeriod') + 1
+        except ec.CodesInternalError:
+            LOGGER.warning("Track %s has no defined timePeriod."
+                            "Track is discarded.", sid)
+            return None
 
-        # setup parsers and querents
-        npparser = pybufrkit.dataquery.NodePathParser()
-        data_query = pybufrkit.dataquery.DataQuerent(npparser).query
+        # ensemble members number
+        ens_no = ec.codes_get_array(bufr, "ensembleMemberNumber")
+
+        # values at timestep 0
```

Comment:
Is there a reason we don't check the significance value here? I can add it in if useful.
Also ... you have to explain the significance value to me at some point.